Can't @Inject a @ConfigMapping Interface on a Validator

### Describe the bug

I implemented a `@ConfigMapping` Interface to pass configurations via `application.yaml` file, annotated it with `@StaticInitSafe` but I can't build my app when I `@Inject` it on a `@ApplicationScoped` `ConstraintValidator`.

I already was using Configurations with the `@ConfigProperty(name = "configname")` annotation and it works, but I implemented configurations via the interface as I'm understanding that it was deprecated and I needed to mock it.

Am I doing something wrong or is it a bug?

### Expected behavior

_No response_

### Actual behavior

_No response_

### How to Reproduce?

_No response_

### Output of `uname -a` or `ver`

_No response_

### Output of `java -version`

Oracle JDK 11

### GraalVM version (if different from Java)

_No response_

### Quarkus version or git rev

2.7.0.Final

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Apache Maven 3.8.4 (9b656c72d54e5bacbed989b64718c159fe39b537)

### Additional information

_No response_

Sorry, should have done that immediately.

Config class:
```java
@StaticInitSafe
@ConfigMapping(prefix = "wally")
public interface AppConfig {

    @WithName("feature-property")
    FeatureProperty featureProperty();

    interface FeatureProperty {
        @WithDefault("0.5")
        double configPercentage();
    }
}
```

Validator class:
```java
@ApplicationScoped
public class CreateEntityValidator implements ConstraintValidator<CreateEntityValidator , EntityRequest> {
// This way it works
//    @ConfigProperty(name = "wally.feature-property.config-percentage")
//    double configPercentage;

    @Inject
    protected AppConfig appConfig;

    public boolean isValid(final EntityRequest entityRequest,
            final ConstraintValidatorContext constraintValidatorContext) {
        this.isValid = true;
        this.errorMessage = new StringBuilder();
        constraintValidatorContext.disableDefaultConstraintViolation();

        if (teamSpaceMaxCapacity < Math.ceil((double) placeCapacity * appConfig.featureProperty().configPercentage())) {
                this.isValid = false;
            }

        return isValid;
    }
}
``` 

Error:
```
2022-02-23 10:09:42,612 ERROR [io.qua.run.boo.StartupActionImpl] (Quarkus Main Thread) Error running Quarkus: java.lang.reflect.InvocationTargetException
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at io.quarkus.runner.bootstrap.StartupActionImpl$1.run(StartupActionImpl.java:103)
	at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: java.lang.ExceptionInInitializerError
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:490)
	at java.base/java.lang.Class.newInstance(Class.java:584)
	at io.quarkus.runtime.Quarkus.run(Quarkus.java:66)
	at io.quarkus.runtime.Quarkus.run(Quarkus.java:41)
	at io.quarkus.runtime.Quarkus.run(Quarkus.java:120)
	at io.quarkus.runner.GeneratedMain.main(Unknown Source)
	... 6 more
Caused by: java.lang.RuntimeException: Failed to start quarkus
	at io.quarkus.runner.ApplicationImpl.<clinit>(Unknown Source)
	... 15 more
Caused by: javax.validation.ValidationException: HV000032: Unable to initialize net.bmwgroup.wally.domain.validation.CreateBookingValidator_ClientProxy.
```

Sorry, should have done that immediately.

Config class:
```java
@StaticInitSafe
@ConfigMapping(prefix = "app")
public interface AppConfig {

    @WithName("feature-property")
    FeatureProperty featureProperty();

    interface FeatureProperty {
        @WithDefault("0.5")
        double configPercentage();
    }
}
```

Validator class:
```java
@ApplicationScoped
public class CreateEntityValidator implements ConstraintValidator<CreateEntityValidator , EntityRequest> {
// This way it works
//    @ConfigProperty(name = "app.feature-property.config-percentage")
//    double configPercentage;

    @Inject
    protected AppConfig appConfig;

    public boolean isValid(final EntityRequest entityRequest,
            final ConstraintValidatorContext constraintValidatorContext) {
        this.isValid = true;
        this.errorMessage = new StringBuilder();
        constraintValidatorContext.disableDefaultConstraintViolation();

        if (teamSpaceMaxCapacity < Math.ceil((double) placeCapacity * appConfig.featureProperty().configPercentage())) {
                this.isValid = false;
            }

        return isValid;
    }
}
``` 

Error:
```
2022-02-23 10:09:42,612 ERROR [io.qua.run.boo.StartupActionImpl] (Quarkus Main Thread) Error running Quarkus: java.lang.reflect.InvocationTargetException
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at io.quarkus.runner.bootstrap.StartupActionImpl$1.run(StartupActionImpl.java:103)
	at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: java.lang.ExceptionInInitializerError
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:490)
	at java.base/java.lang.Class.newInstance(Class.java:584)
	at io.quarkus.runtime.Quarkus.run(Quarkus.java:66)
	at io.quarkus.runtime.Quarkus.run(Quarkus.java:41)
	at io.quarkus.runtime.Quarkus.run(Quarkus.java:120)
	at io.quarkus.runner.GeneratedMain.main(Unknown Source)
	... 6 more
Caused by: java.lang.RuntimeException: Failed to start quarkus
	at io.quarkus.runner.ApplicationImpl.<clinit>(Unknown Source)
	... 15 more
Caused by: javax.validation.ValidationException: HV000032: Unable to initialize net.domain.validation.CreateBookingValidator_ClientProxy.
```

--- a/extensions/arc/deployment/src/main/java/io/quarkus/arc/deployment/ConfigBuildStep.java
+++ b/extensions/arc/deployment/src/main/java/io/quarkus/arc/deployment/ConfigBuildStep.java
@@ -289,7 +289,7 @@ public class ConfigBuildStep {
         Map<Type, ConfigClassBuildItem> configMappingTypes = configClassesToTypesMap(configClasses, MAPPING);
         Set<ConfigClassBuildItem> configMappings = new HashSet<>();
         for (InjectionPointInfo injectionPoint : beanRegistration.getInjectionPoints()) {
-            Type type = Type.create(injectionPoint.getType().name(), Type.Kind.CLASS);
+            Type type = Type.create(injectionPoint.getRequiredType().name(), Type.Kind.CLASS);
             ConfigClassBuildItem configClass = configMappingTypes.get(type);
             if (configClass != null) {
                 configMappings.add(configClass);
@@ -331,7 +331,7 @@ public class ConfigBuildStep {
                 continue;
             }
 
-            Type type = Type.create(injectionPoint.getType().name(), Type.Kind.CLASS);
+            Type type = Type.create(injectionPoint.getRequiredType().name(), Type.Kind.CLASS);
             ConfigClassBuildItem configClass = configPropertiesTypes.get(type);
             if (configClass != null) {
                 configProperties.add(configClass);
@@ -367,7 +367,7 @@ public class ConfigBuildStep {
         Map<Type, ConfigClassBuildItem> configMappingTypes = configClassesToTypesMap(configClasses, MAPPING);
         Set<ConfigMappingBuildItem> toRegister = new HashSet<>();
         for (InjectionPointInfo injectionPoint : validationPhase.getContext().getInjectionPoints()) {
-            Type type = Type.create(injectionPoint.getType().name(), Type.Kind.CLASS);
+            Type type = Type.create(injectionPoint.getRequiredType().name(), Type.Kind.CLASS);
             ConfigClassBuildItem configClass = configMappingTypes.get(type);
             if (configClass != null) {
                 AnnotationTarget target = injectionPoint.getTarget();
@@ -422,7 +422,7 @@ public class ConfigBuildStep {
         for (InjectionPointInfo injectionPoint : validationPhase.getContext().getInjectionPoints()) {
             AnnotationInstance properties = injectionPoint.getRequiredQualifier(MP_CONFIG_PROPERTIES_NAME);
             if (properties != null) {
-                Type type = Type.create(injectionPoint.getType().name(), Type.Kind.CLASS);
+                Type type = Type.create(injectionPoint.getRequiredType().name(), Type.Kind.CLASS);
                 ConfigClassBuildItem configClass = configPropertiesTypes.get(type);
                 if (configClass != null) {
                     AnnotationValue annotationPrefix = properties.value("prefix");

--- a/extensions/arc/deployment/src/test/java/io/quarkus/arc/test/config/ConfigMappingInstanceTest.java
+++ b/extensions/arc/deployment/src/test/java/io/quarkus/arc/test/config/ConfigMappingInstanceTest.java
@@ -1,0 +1,55 @@
+package io.quarkus.arc.test.config;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+
+import javax.enterprise.inject.Instance;
+import javax.inject.Inject;
+
+import org.eclipse.microprofile.config.inject.ConfigProperties;
+import org.eclipse.microprofile.config.inject.ConfigProperty;
+import org.jboss.shrinkwrap.api.asset.StringAsset;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import io.quarkus.test.QuarkusUnitTest;
+import io.smallrye.config.ConfigMapping;
+import io.smallrye.config.SmallRyeConfig;
+import io.smallrye.config.WithName;
+
+public class ConfigMappingInstanceTest {
+    @RegisterExtension
+    static final QuarkusUnitTest TEST = new QuarkusUnitTest()
+            .withApplicationRoot((jar) -> jar
+                    .addAsResource(new StringAsset("config.my.prop=1234\n"), "application.properties"));
+
+    @ConfigMapping(prefix = "config")
+    public interface MyConfigMapping {
+        @WithName("my.prop")
+        String myProp();
+    }
+
+    @ConfigProperties(prefix = "config")
+    public static class MyConfigProperties {
+        @ConfigProperty(name = "my.prop")
+        String myProp;
+    }
+
+    @Inject
+    SmallRyeConfig config;
+    @Inject
+    Instance<MyConfigMapping> myConfigMapping;
+    @Inject
+    @ConfigProperties
+    Instance<MyConfigProperties> myConfigProperties;
+
+    @Test
+    void configMapping() {
+        MyConfigMapping configMapping = config.getConfigMapping(MyConfigMapping.class);
+        assertNotNull(configMapping);
+        assertEquals("1234", configMapping.myProp());
+
+        assertEquals("1234", myConfigMapping.get().myProp());
+        assertEquals("1234", myConfigProperties.get().myProp);
+    }
+}

